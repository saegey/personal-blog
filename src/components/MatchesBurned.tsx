import { useState } from 'react'
import { Box, Flex, Text, Button } from 'theme-ui'

import Modal from './Modal'

const MatchesHeading = () => {
  return (
    <Flex sx={{ height: ['35px', '35px', '35px'], marginTop: '20px' }}>
      <Box sx={{ width: '34%' }}>
        <Text as="p" variant="resultsItem">
          <strong>Average Power</strong>
        </Text>
      </Box>
      <Box sx={{ width: '33%', textAlign: 'center' }}>
        <Text as="p" variant="resultsItem">
          <strong>Total Joules</strong>
        </Text>
      </Box>
      <Box sx={{ width: '33%', textAlign: 'right' }}>
        <Text as="p" variant="resultsItem">
          <strong>Total time</strong>
        </Text>
      </Box>
    </Flex>
  )
}
type ItemProps = {
  index: number
  d: {
    averagePower: number
    totalJoules: number
    totalTime: number
    vals: number[]
    startTime: string
  }
}
const MatchesItem = ({ index, d }: ItemProps) => {
  return (
    <Flex key={`match${index}`} sx={{ height: ['35px', '35px', '35px'] }}>
      <Box sx={{ width: '34%' }}>
        <Text as="p" variant="resultsItem">
          {d.averagePower} watts
        </Text>
      </Box>
      <Box sx={{ width: '33%', textAlign: 'center' }}>
        <Text as="p" variant="resultsItem">
          {(d.totalJoules / 1000).toFixed(2)} kJ
        </Text>
      </Box>
      <Box sx={{ width: '33%', textAlign: 'right' }}>
        <Text as="p" variant="resultsItem">
          {d.totalTime} sec
        </Text>
      </Box>
    </Flex>
  )
}

type Props = {
  data: [
    {
      averagePower: number
      totalJoules: number
      totalTime: number
      vals: number[]
      startTime: string
    }
  ]
}

export const MatchesBurned = ({ data }: Props) => {
  const [isModalOpen, openModal] = useState(false)

  return (
    <>
      {isModalOpen && (
        <Modal modalOpen={openModal} headerText={"Matches Burned"}>
          <>
            <MatchesHeading />
            {data.map((d, index: number) => {
              return <MatchesItem d={d} index={index} key={index} />
            })}
          </>
        </Modal>
      )}
      <Box>
        <Flex>
          <Box sx={{ marginBottom: ['10px', '0px', '0px'] }}>
            <Text as="h2" variant="resultsHeading">
              Matches Burned
            </Text>
          </Box>
          <Box sx={{ marginLeft: 'auto' }}>
            <Button
              onClick={() => {
                openModal(true)
              }}
            >
              View All
            </Button>
          </Box>
        </Flex>
        <MatchesHeading />
        {data.slice(0, 5).map((d, index: number) => {
          return <MatchesItem d={d} index={index} />
        })}
      </Box>
    </>
  )
}
