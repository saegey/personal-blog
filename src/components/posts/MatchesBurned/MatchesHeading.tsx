import { Flex, Box, Text } from 'theme-ui'

const MatchesHeading = () => {
  return (
    <Flex sx={{ height: ['35px', '35px', '35px'], marginTop: '20px' }}>
      <Box sx={{ width: '34%' }}>
        <Text as="p" variant="resultsItem">
          <strong>Average Power</strong>
        </Text>
      </Box>
      <Box sx={{ width: '33%', textAlign: 'center' }}>
        <Text as="p" variant="resultsItem">
          <strong>Total Joules</strong>
        </Text>
      </Box>
      <Box sx={{ width: '33%', textAlign: 'right' }}>
        <Text as="p" variant="resultsItem">
          <strong>Total time</strong>
        </Text>
      </Box>
    </Flex>
  )
}

export default MatchesHeading
