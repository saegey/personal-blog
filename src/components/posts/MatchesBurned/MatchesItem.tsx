import { Flex, Box, Text } from 'theme-ui'

type ItemProps = {
  index: number
  d: {
    averagePower: number
    totalJoules: number
    totalTime: number
    vals: number[]
    startTime: string
  }
}

const MatchesItem = ({ index, d }: ItemProps) => {
  return (
    <Flex key={`match${index}`} sx={{ height: ['35px', '35px', '35px'] }}>
      <Box sx={{ width: '34%' }}>
        <Text as="p" variant="resultsItem">
          {d.averagePower} watts
        </Text>
      </Box>
      <Box sx={{ width: '33%', textAlign: 'center' }}>
        <Text as="p" variant="resultsItem">
          {(d.totalJoules / 1000).toFixed(2)} kJ
        </Text>
      </Box>
      <Box sx={{ width: '33%', textAlign: 'right' }}>
        <Text as="p" variant="resultsItem">
          {d.totalTime} sec
        </Text>
      </Box>
    </Flex>
  )
}

export default MatchesItem
