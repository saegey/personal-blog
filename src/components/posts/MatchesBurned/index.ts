import { MatchesBurned } from './MatchesBurned';

export default MatchesBurned
