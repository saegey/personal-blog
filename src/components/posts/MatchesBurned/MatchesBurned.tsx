import React from 'react'
import { Box, Flex, Text } from 'theme-ui'

import FullScreenIcon from '../../icons/FullScreenIcon'
import Modal from '../../Modal'
import MatchesHeading from './MatchesHeading'
import MatchesItem from './MatchesItem'

type Props = {
  data: Array<{
    averagePower: number
    totalJoules: number
    totalTime: number
    vals?: number[]
    startTime?: string
  }>
}

export const MatchesBurned = ({ data }: Props) => {
  const [isModalOpen, openModal] = React.useState(false)

  return (
    <>
      {isModalOpen && (
        <Modal
          modalOpen={openModal}
          headerText={'Matches Burned'}
          testId={'matches-modal'}
        >
          <>
            <MatchesHeading />
            {data.map((d, index: number) => {
              return <MatchesItem d={d} index={index} key={index} />
            })}
          </>
        </Modal>
      )}
      <Box
        sx={{
          maxWidth: '690px',
          margin: '60px auto',
          background: 'muted',
          padding: '30px',
          borderRadius: '5px',
        }}
      >
        <Flex>
          <Box sx={{ marginBottom: '20px' }}>
            <Text as="h2" variant="resultsHeading">
              Matches Burned
            </Text>
          </Box>
          <Box sx={{ marginLeft: 'auto' }}>
            <Box as="button"
              data-testid={'matches-burned-fullscreen'}
              sx={{
                width: '40px',
                right: '0',
                top: '0',
                zIndex: 0,
              }}
              onClick={() => {
                openModal(true)
              }}
            >
              <FullScreenIcon color="primary" />
            </Box>
          </Box>
        </Flex>
        <MatchesHeading />
        {data.slice(0, 5).map((d, index: number) => {
          return <MatchesItem d={d} key={`matches-${index}`} index={index} />
        })}
      </Box>
    </>
  )
}
