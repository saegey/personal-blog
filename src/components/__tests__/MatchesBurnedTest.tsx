/**
 * @jest-environment jsdom
 */
import renderer from 'react-test-renderer'
import { render, screen, fireEvent } from '@testing-library/react'
import { ThemeProvider } from 'theme-ui'

import { MatchesBurned } from '../MatchesBurned'
import { theme } from './__test-utils__'

describe('Matches Burned', () => {
  it('renders', () => {
    const json = renderer
      .create(
        <ThemeProvider theme={theme}>
          <MatchesBurned
            data={[
              {
                averagePower: 200,
                totalJoules: 1000,
                totalTime: 60,
                vals: [0, 2, 3],
                startTime: '3443434',
              },
            ]}
          />
        </ThemeProvider>
      )
      .toJSON()
    expect(json).toMatchSnapshot()
  })

  it('opens a modal if view all is clicked', () => {
    render(
      <ThemeProvider theme={theme}>
        <MatchesBurned
          data={[
            {
              averagePower: 200,
              totalJoules: 1000,
              totalTime: 60,
              vals: [0, 2, 3],
              startTime: '3443434',
            },
          ]}
        />
      </ThemeProvider>
    )

    const viewAll = screen.getByRole('button')
    fireEvent.click(viewAll)
    expect(screen.queryAllByTestId('matchesModal').length).toEqual(1)
  })
})
