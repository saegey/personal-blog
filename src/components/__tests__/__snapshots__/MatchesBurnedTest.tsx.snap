// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches Burned renders 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-bottom: 10px;
}

@media screen and (min-width: 40em) {
  .emotion-2 {
    margin-bottom: 0px;
  }
}

@media screen and (min-width: 52em) {
  .emotion-2 {
    margin-bottom: 0px;
  }
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: auto;
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary);
  border: 0;
  border-radius: 4px;
  color: var(--theme-ui-colors-background);
  background-color: var(--theme-ui-colors-primary);
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 35px;
  margin-top: 20px;
}

@media screen and (min-width: 40em) {
  .emotion-6 {
    height: 35px;
  }
}

@media screen and (min-width: 52em) {
  .emotion-6 {
    height: 35px;
  }
}

.emotion-7 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 34%;
}

.emotion-9 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 33%;
  text-align: center;
}

.emotion-11 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 33%;
  text-align: right;
}

.emotion-13 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 35px;
}

@media screen and (min-width: 40em) {
  .emotion-13 {
    height: 35px;
  }
}

@media screen and (min-width: 52em) {
  .emotion-13 {
    height: 35px;
  }
}

.emotion-14 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 34%;
}

.emotion-16 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 33%;
  text-align: center;
}

.emotion-18 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 33%;
  text-align: right;
}

<div
  className="emotion-0"
>
  <div
    className="emotion-1"
  >
    <div
      className="emotion-2"
    >
      <h2
        className="emotion-0"
      >
        Matches Burned
      </h2>
    </div>
    <div
      className="emotion-4"
    >
      <button
        className="emotion-5"
        onClick={[Function]}
      >
        View All
      </button>
    </div>
  </div>
  <div
    className="emotion-6"
  >
    <div
      className="emotion-7"
    >
      <p
        className="emotion-0"
      >
        <strong>
          Average Power
        </strong>
      </p>
    </div>
    <div
      className="emotion-9"
    >
      <p
        className="emotion-0"
      >
        <strong>
          Total Joules
        </strong>
      </p>
    </div>
    <div
      className="emotion-11"
    >
      <p
        className="emotion-0"
      >
        <strong>
          Total time
        </strong>
      </p>
    </div>
  </div>
  <div
    className="emotion-13"
  >
    <div
      className="emotion-14"
    >
      <p
        className="emotion-0"
      >
        200
         watts
      </p>
    </div>
    <div
      className="emotion-16"
    >
      <p
        className="emotion-0"
      >
        1.00
         kJ
      </p>
    </div>
    <div
      className="emotion-18"
    >
      <p
        className="emotion-0"
      >
        60
         sec
      </p>
    </div>
  </div>
</div>
`;
